test/ruby/test_gc.rb

```diff
@@ -315,6 +315,15 @@ def test_gc_parameter
     }
     assert_normal_exit("exit", "[ruby-core:39777]", :child_env => env)
 
+    env = {}
+    GC.stat_heap.each do |_, s|
+      env["RUBY_GC_HEAP_INIT_SIZE_#{s[:slot_size]}_SLOTS"] = "200000"
+    end
+    assert_normal_exit("exit", "", :child_env => env)
+
+    env["RUBY_GC_HEAP_INIT_SLOTS"] = "100000"
+    assert_normal_exit("exit", "", :child_env => env)
```

Comment:
Can you also add a test to ensure that it doesn't crash with a small number (like 0)?

Reply:
Done.